api/server/router/container/container_routes.go

```diff
@@ -446,10 +428,6 @@ func (s *containerRouter) wsContainersAttach(ctx context.Context, w http.Respons
 	}
 	containerName := vars["name"]
 
-	if !s.backend.Exists(containerName) {
-		return derr.ErrorCodeNoSuchContainer.WithArgs(containerName)
```

Comment:
Yep, just haven't had a chance to look at how to fix it.
Also, the websocket version is really a hijacker too, not sure if I can coalesce that.